src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs

```diff
@@ -385,15 +385,15 @@ public static Expression GetColValueExpr(ParameterExpression readerParamExpr, in
                     callExpr = Expression.Call(readerParamExpr, GetterForType(underlyingType), iConstant);
                 }
                 Expression colValueExpr;
-                if (!canBeNull) {
+                var isTypeWithCreateMethod = CreateMethodOfTypeWithCreateMethod(underlyingType, out var methodInfo);
+                if (!canBeNull && !isTypeWithCreateMethod) {
                     colValueExpr = GetCastExpression(callExpr, type);
                 }
-                else if (CreateMethodOfTypeWithCreateMethod(underlyingType, out var methodInfo)) {
+                else if (isTypeWithCreateMethod) {
                     var test = Expression.Call(readerParamExpr, IsDBNullMethod, iConstant);
-                    var paramType = methodInfo.GetParameters()[0].ParameterType;
-                    var ifDbNull = Expression.Default(paramType);
-                    var ifNotDbNull = Expression.Convert(GetCastExpression(callExpr, paramType), paramType);
-                    colValueExpr = Expression.Call(methodInfo, Expression.Condition(test, ifDbNull, ifNotDbNull));
+                    var ifDbNull = Expression.Default(type);

```

Comment:
dus mag geen value type zijn?

Voelt logischer denk ik om deze hele logica binnen `GetCastExpression` te doen; dan krijg je de nette `null`-lifting vanzelf erbij.  Evt dan hernoemen; maar het *is* ook een soort cast.